pkg/lib/bundlev2/generate.go

```diff
@@ -0,0 +1,144 @@
+package bundlev2
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+	"k8s.io/kubectl/pkg/util/slice"
+)
+
+const SynthesizedCSVFileName string = "olm.clusterserviceversion.yaml"
+
+// GenerateFunc generates a CSV for the bundle based on the resource within the bundle directory.
+// The bundleDir points to where the bundle is located, outputDir is where bundle and synthesized CSV is copy/write to.
+// If overwrite is set to true, outputDir is reloaded with the exact content of bundleDir plus a synthesized CSV
+// and inplace generation deletes the CSV if exists.
+func GenerateFunc(bundleDir, outputDir string, overwrite bool) error {
```

Comment:
```suggestion
func Generate(bundleDir, outputDir string, overwrite bool) error {
```

Reply:
we use the "functional option" pattern a lot elsewhere in this codebase, you might think about using that for outputDir and overwrite